### PR TITLE
feat(source): add webhook source

### PR DESCRIPTION
**Pull request by WanYixian**
## Description

This is the follow-up of https://github.com/risingwavelabs/risingwave-docs-legacy/pull/2762

## Related Code PR
https://github.com/risingwavelabs/risingwave/pull/19272

### CODE REVIEW

File: integrations/sources/webhook.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+title: "Ingest data from Webhook"
```
Comment (hengm3467) [inline integrations/sources/webhook.mdx]: Let's use "webhooks" here. It's a general technology / mechanism that do not need to be captalized.

File: integrations/sources/webhook.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+This feature is only available in the premium edition of RisingWave. The premium edition offers additional advanced features and capabilities beyond the free and community editions. If you have any questions about upgrading to the premium edition, please contact our sales team at [sales@risingwave-labs.com](mailto:sales@risingwave-labs.com).
+</Tip>
+
+<Note>
+**PUBLIC PREVIEW**
+
+This feature is in the public preview stage, meaning it's nearing the final product but is not yet fully stable. If you encounter any issues or have feedback, please contact us through our [Slack channel](https://www.risingwave.com/slack). Your input is valuable in helping us improve the feature. For more information, see our [Public preview feature list](/product-lifecycle/#features-in-the-public-preview-stage).
+</Note>
+
+## Creating a Webhook table in RisingWave
```
Comment (hengm3467) [inline integrations/sources/webhook.mdx]:
```suggestion
## Creating a webhook table in RisingWave
```

File: integrations/sources/webhook.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+| `CREATE SECRET` | Securely stores a secret value in RisingWave for request validation. |
+| `CREATE TABLE` | Defines a table with a JSONB column to store webhook payload data. |
+| `connector` | Configures the table to accept incoming HTTP webhook requests |
+| `VALIDATE SECRET...AS...` | Authenticates requests using the stored secret and signature comparison. |
+| `secure_compare()` | Validates requests by matching the header signature against the computed signature, ensuring only authenticated requests are processed. Note `secure_compare(...)` is the only supported validation function for webhook tables. |
+| `header_of_signature` | Specifies which HTTP header contains the incoming signature. |
+| `signature_generation_expressions` | Expression to compute the expected signature using the secret and payload. |
+
+
+## Supported Webhook sources and authentication methods
```
Comment (hengm3467) [inline integrations/sources/webhook.mdx]:
```suggestion
## Supported webhook sources and authentication methods
```

File: integrations/sources/webhook.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+|Webhook source|Authentication methods|
+| :-- | :-- |
+|GitHub| SHA-1 HMAC, SHA-256 HMAC |
+|Rudderstack| Bearer Token |
+|Segment| SHA-1 HMAC |
+|AWS EventBridge| Bearer Token |
+|HubSpot| API Key, Signature V2 |
+
+<Note>While only the above sources have been thoroughly tested, RisingWave can support additional Webhook sources and authentication methods. You can integrate other services using similar configurations, though they haven't been officially verified.</Note>
```
Comment (hengm3467) [inline integrations/sources/webhook.mdx]:
```suggestion
<Note>While only the above sources have been thoroughly tested, RisingWave can support additional webhook sources and authentication methods. You can integrate other services using similar configurations.</Note>
```

**Review by hengm3467 — APPROVED** [review]
Added some suggestions. Otherwise LGTM. Thanks.

File: integrations/sources/github-webhook.mdx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+
+| Parameter or clause | Description |
+| :--------- | :-----------|
+| `data JSONB` | Defines the name of column to store the JSON payload from the webhook. Currently, only `JSONB` type is supported for webhook tables. |
+| `headers->>'...'` | Extracts the signature provided by GitHub in the `x-hub-signature-256` HTTP header. <br/> <br/> In `secure_compare()` function, the whole HTTP header is interpreted as a JSONB object, and you can access the header value using the `->>` operator, but only the lower-case header names in the `->>` operator, otherwise the verification will fail. |
+|`'sha256=' \|\| encode(...)` | Computes the expected signature. In the example above, it generates an `HMAC SHA-256` hash of the payload (`data`) using the secret (`test_secret`), encodes it in hexadecimal, and prefixes it with `sha256=`.|
+| `secure_compare(...)` | Validates requests by matching the header signature against the computed signature, ensuring only authenticated requests are processed. The `secure_compare()` function compares two strings in a fixed amount of time, regardless of whether they are equal or not, ensuring that the comparison is secure and resistant to timing attacks.
```
Comment (KeXiangWang) [inline integrations/sources/github-webhook.mdx]:
```suggestion
| `secure_compare(...)` | Validates requests by matching the header signature against the computed signature, ensuring only authenticated requests are processed. The `secure_compare()` function compares two strings in a fixed amount of time, regardless of whether they are equal or not, ensuring that the comparison is secure and resistant to timing attacks. |
```

**Pull request merged**